apps/testing/src/main/java/com/akto/test_editor/strategy/Strategy.java

```diff
@@ -0,0 +1,70 @@
+package com.akto.test_editor.strategy;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import com.akto.dao.ApiCollectionsDao;
+import com.akto.dao.context.Context;
+import com.akto.dto.ApiCollection;
+import com.akto.dto.testing.GenericTestResult;
+import com.akto.dto.testing.TestResult;
+import com.akto.dto.testing.YamlTestResult;
+import com.akto.log.LoggerMaker;
+import com.akto.log.LoggerMaker.LogDb;
+import com.akto.test_editor.Utils;
+
+public class Strategy {
+    
+    private static final String AKTO_DISCOVERED_APIS_COLLECTION = "shadow_apis";
+    private static final int AKTO_DISCOVERED_APIS_COLLECTION_ID = 1333333333;
+    private static final LoggerMaker loggerMaker = new LoggerMaker(Strategy.class);
+
+    public static void triggerStrategyInstructions(com.akto.dto.test_editor.Strategy strategy, YamlTestResult attempts) {
+        
+        if (strategy == null || !strategy.getInsertVulnApi()) {
+            return;
+        }
+
+        String harPayload = null;
+
+        try {
+            for (GenericTestResult testRes: attempts.getTestResults()) {
+                if (!testRes.getVulnerable()) {
+                    continue;
+                }
+                if (testRes instanceof TestResult) {
+                    TestResult res = (TestResult) testRes;
+                    String message = res.getMessage();
+                    try {
+                        harPayload = Utils.convertToHarPayload(message, Context.accountId.get(), Context.now(), "", "HAR");
+                    } catch (Exception e) {
+                        loggerMaker.errorAndAddToDb("Error while converting attempt to har payload " + e.getMessage(), LogDb.TESTING);
```

Comment:
We can remove the second param, after changing the function call of the loggermaker fn.